### ISSUE TITLE
Support widget areas

### ISSUE THREAD

**Issue by jpommerening** [LaxarJS/laxar-vue-adapter#3]
I found this code lying in my local copy of the `laxar-vue-adapter`:

```js
mounted() {
   const areaHelper = services.axAreaHelper;
   const slots = collectSlots(this._vnode);

   Object.keys(slots).forEach(key => {
      areaHelper.register(key, slots[key].elm);
   });
}

function collectSlots( $vnode, slots = {} ) {
   if( $vnode.data && $vnode.data.slot ) {
      slots[ $vnode.data.slot ] = $vnode;
   }
   else if( $vnode.children ) {
      $vnode.children.forEach($child => collectSlots($child, slots));
   }
   return slots;
}
```

The mount callback was part of the mixins added by the adapter. When mounted, the code traverses the component's v-dom to register the corresponding elements with the area helper.

IIRC this worked quite seamlessly. A nicer way to implement _this_ approach would be to wrap the `render()` function (which returns the v-dom), collect the slots there and then publish the areas when mounted.

Or we could just create a custom `ax-widget-area` directive ;)

**Comment by x1B** [comment]
TBH I'd advocate for the directive, which seems like the right tool for this. Assuming we can access the areaHelper there without too much hassle (maybe pass it in like the react adapter does).

**Comment by x1B** [comment]
Implemented on `master` (`v2.0.0`).

### Upgrade Information

The laxar-vue-adapter now provides an implicit component to all widgets and controls: use `<ax-widget-area name="x" />` to add a LaxarJS widget area whose content can then be controlled from the page.
Consult the README for details.
